### PR TITLE
Add dynamic reconfigure to ball_filter

### PR DESCRIPTION
**Pull request by jaagut**
## Proposed changes
Add dynamic reconfigure to ball_filter

## Necessary checks
- [x] Run `catkin build`
~- [ ] Write documentation~
~- [ ] Create issues for future work~
- [x] Test on your machine
~- [ ] Test on the robot~
- [x] Put the PR on our Project board

### CODE REVIEW

File: bitbots_ball_filter/cfg/BallFilter.cfg [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+group_ROS.add("ball_subscribe_topic", str_t, 0, "", None)
+group_ROS.add("ball_position_publish_topic", str_t, 0, "", None)
+group_ROS.add("ball_movement_publish_topic", str_t, 0, "", None)
+group_ROS.add("ball_publish_topic", str_t, 0, "", None)
+group_ROS.add("filter_frame", str_t, 0, "", None)
+
+
+group_filter.add("filter_rate", int_t, 0, "Filtering rate in Hz", min=0, max=255)
+group_filter.add("velocity_reduction", double_t, 0, "Velocity reduction (per axis) factor of the ball per second", min=0, max=1.0)
+group_filter.add("process_noise_variance", double_t, 0, "", min=0, max=1.0)
```
Comment (Flova) [inline bitbots_ball_filter/cfg/BallFilter.cfg]: Please add a parameter desciption

Comment (Flova) [reply bitbots_ball_filter/cfg/BallFilter.cfg]: And evaluate the value range

File: bitbots_ball_filter/cfg/BallFilter.cfg [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+gen = ParameterGenerator()
+
+group_ROS= gen.add_group("ROS", type="tab")
+group_filter= gen.add_group("Filter", type="tab")
+
+group_ROS.add("ball_subscribe_topic", str_t, 0, "", None)
+group_ROS.add("ball_position_publish_topic", str_t, 0, "", None)
+group_ROS.add("ball_movement_publish_topic", str_t, 0, "", None)
+group_ROS.add("ball_publish_topic", str_t, 0, "", None)
+group_ROS.add("filter_frame", str_t, 0, "", None)
```
Comment (Flova) [inline bitbots_ball_filter/cfg/BallFilter.cfg]: Don't dyn reconf the filter frame. Instead, set it in the launch file depending of the environment variable

**Review by Flova — CHANGES_REQUESTED** [review]

File: bitbots_ball_filter/cfg/BallFilter.cfg (hunk 2 above)
Comment (NFiedler) [reply bitbots_ball_filter/cfg/BallFilter.cfg]: which environment variable?

File: bitbots_ball_filter/cfg/BallFilter.cfg (hunk 1 above)
Comment (jaagut) [reply bitbots_ball_filter/cfg/BallFilter.cfg]: @NFiedler can you describe, what this param does, please?

File: bitbots_ball_filter/cfg/BallFilter.cfg [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+group_ROS.add("ball_subscribe_topic", str_t, 0, "", None)
+group_ROS.add("ball_position_publish_topic", str_t, 0, "", None)
+group_ROS.add("ball_movement_publish_topic", str_t, 0, "", None)
+group_ROS.add("ball_publish_topic", str_t, 0, "", None)
+group_ROS.add("use_frame", str_t, 0, "Choose a frame to use", "odom", edit_method=frames_chooser_enum)
+
+
+group_filter.add("filter_rate", int_t, 0, "Filtering rate in Hz", min=0, max=255)
+group_filter.add("velocity_reduction", double_t, 0, "Velocity reduction (per axis) factor of the ball per second", min=0, max=1.0)
+group_filter.add("process_noise_variance", double_t, 0, "", min=0, max=1.0)
```
Comment (NFiedler) [inline bitbots_ball_filter/cfg/BallFilter.cfg]:
```suggestion
group_filter.add("process_noise_variance", double_t, 0, "Noise which is added to the estimated position without new measurements", min=0, max=1.0)
```

**Comment by NFiedler** [comment]
Can we merge this now, @Flova ? We tested it on CL03 and it works.

**Comment by Flova** [comment]
I also tested this on the cl05

**Review by Flova — APPROVED** [review]

**Pull request merged**